core/src/main/java/org/kohsuke/stapler/RequestImpl.java

```diff
@@ -1056,15 +1062,15 @@ private void parseMultipartFormData() throws ServletException {
 
         parsedFormData = new HashMap<>();
         parsedFormDataFormFields = new HashMap<>();
-        ServletFileUpload upload;
+        JavaxServletFileUpload<DiskFileItem, DiskFileItemFactory> upload;
         File tmpDir;
         try {
             tmpDir = Files.createTempDirectory("jenkins-stapler-uploads").toFile();
         } catch (IOException e) {
             throw new ServletException("Error creating temporary directory", e);
         }
         tmpDir.deleteOnExit();
-        upload = new ServletFileUpload(new DiskFileItemFactory(DiskFileItemFactory.DEFAULT_SIZE_THRESHOLD, tmpDir));
+        upload = new JavaxServletDiskFileUpload(DiskFileItemFactory.builder().setFile(tmpDir).get());
```

Comment:
The new builder API allows us to omit the default size threshold without any change in behavior. I verified the builder's default is the same number as the old default that we were explicitly passing in.